upgrade-2.x.sh

```diff
@@ -1250,6 +1282,43 @@ FETCHED_SLUG=$(CURL_CA_BUNDLE="${TMPCRT}" ${CURL} -H "Authorization: Bearer ${AP
 SLUG=${FORCED_SLUG:-$FETCHED_SLUG}
 HOST_OS_VERSION=${META_BALENA_VERSION:-${VERSION_ID}}
 
+# Must query for target version and target image, which includes registry endpoint,
+# if started from app UUID.
+if [ -n "$app_uuid" ]; then
+    # Retrieve release and target image information for the provided app UUID
+    # and release commit. Verify that the release succeeded and has not been invalidated.
+    # We expect at most a single release. Catch command failure for handling below.
+    query_res=$(CURL_CA_BUNDLE="${TMPCRT}" ${CURL} \
+        -H "Content-Type: application/json" -H "Authorization: Bearer ${APIKEY}" \
+        "${API_ENDPOINT}/v6/release?\$select=raw_version&\$expand=contains__image(\$expand=image(\$select=is_stored_at__image_location,content_hash))&\$filter=commit%20eq%20%27${release_commit}%27%20and%20(belongs_to__application/any(bta:bta/is_host%20and%20bta/uuid%20eq%20%27${app_uuid}%27))%20and%20status%20eq%20'success'%20and%20is_invalidated%20eq%20false" || echo "fail")
```

Comment:
Added more comprehensive error checking on release query. Notice `||` at end and how we look for "fail" below. Also be sure the JSON response includes a row.